### DISCUSSION TITLE
格式检查相关问题（abstract字体、参考文献行间距等）

### DISCUSSION THREAD

**Discussion by yaoyichloe** [stone-zeng/fduthesis#326] [Q&A]
格式检查要求改abstract的字体，该怎么解决呢

**Comment by sikouhjw** [comment]
在导言区添加
```tex
\ctexset{
  chapter/format += \fontspec{Times New Roman}
}
```
注意，该命令会让所有 `\chapter` 中的西文字体均为 TNR。如果只改动 abstract 而不改动其他章节是不合理的。

**Reply by slzhoufdu2024** [reply]
您好，请问这里如果只对abstract作调整，该怎么修改呢？

**Reply by sikouhjw** [reply]
> 您好，请问这里如果只对abstract作调整，该怎么修改呢？

你不觉得这个需求是矛盾的吗？abstract 是不编号的 chapter，我不认为需要区分。如果需要，我相信是提意见的人的审美问题。

当然，解决方案是存在的，即在 abstract 后的位置添加
```tex
\ctexset{
  chapter/format += \fontspec{Arial}
}
```

**Comment by yaoyichloe** [comment]
谢谢，问题解决了！
abstract的字体确实变了，并且一级标题的字体也变成了Times New Roman字体！
但是目录中一级标题的英文字体还是没有变化，正文中英文字体确实变了
请问这个问题可以解决嘛？






------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "stone-zeng/fduthesis"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年3月17日(星期天) 下午5:28
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [stone-zeng/fduthesis] 英语摘要中的abstract字体怎么修改呢？如何改成Times New Roman字体呢 (Discussion #326)





 
在导言区添加
 \ctexset{   chapter/format += \fontspec{Times New Roman} }
 
注意，该命令会让所有 \chapter 中的西文字体均为 TNR。如果只改动 abstract 而不改动其他章节是不合理的。

**Comment by sikouhjw** [comment]
>  但是目录中一级标题的英文字体还是没有变化，正文中英文字体确实变了 请问这个问题可以解决嘛？

这不是问题，你从来就没有提出这个需求。格式检查是怎么说的？哪个学院的非适用性要求还是统一要求？

**Comment by yaoyichloe** [comment]
您好！


我是信息学院的，格式检查并没有提出我这个问题，只是提到了abstract的字体的问题，通过您给的方法我这个问题已经解决了。
目录中英文符号的字体问题是我自己发现的，并不是格式检查中提出的问题。
格式检查中仅提出了abstract的字体问题，以及参考文献的行间距问题。
请问您有什么办法解决参考文献的行间距不一致的问题嘛？




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "stone-zeng/fduthesis"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年3月17日(星期天) 晚上6:07
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [stone-zeng/fduthesis] 英语摘要中的abstract字体怎么修改呢？如何改成Times New Roman字体呢 (Discussion #326)





  
但是目录中一级标题的英文字体还是没有变化，正文中英文字体确实变了 请问这个问题可以解决嘛？
  
这不是问题，你从来就没有提出这个需求。格式检查是怎么说的？哪个学院的非适用性要求还是统一要求？

**Reply by sikouhjw** [reply]
> 目录中英文符号的字体问题是我自己发现的，并不是格式检查中提出的问题。

首先，既然不是格式检查的问题，那它就不是「问题」。学校认可这种格式，再去修改是无意义的。其次，「chapter 长什么样」与「目录中 chapter 长什么样」是独立的，没有任何关联的。

> 格式检查中仅提出了abstract的字体问题，以及参考文献的行间距问题。 请问您有什么办法解决参考文献的行间距不一致的问题嘛？

1. 如果讨论格式检查的问题，该 issue 标题应与格式检查相关。
2. 请详细描述行间距有什么问题，用文字及图片去描述问题

**Reply by yaoyichloe** [reply]
好的，题目已经修改。
格式检查中提到参考文献行间距不同。
我猜测是因为一个引文不会出现在上下两页，会自动编在一页中，这会导致行间距比较大或者有的行间距比较小。
请问这个问题可以解决嘛

**Reply by sikouhjw** [reply]
在 `\printbibliography` 前面加上 `\setlength{\bibsep}{0pt}` (bibex) 或 `\setlength{\bibitemsep}{0pt}` (biblatex)，参考 https://ask.latexstudio.net/ask/question/7229.html

**Reply by yaoyichloe** [reply]
好的，问题解决了！十分感谢

**Comment by yaoyichloe** [comment]
请问表头的字体不是宋体和Times New Roman，好像是黑体，这个问题可以解决嘛？

**Reply by sikouhjw** [reply]
> 请问表头的字体不是宋体和Times New Roman，好像是黑体，这个问题可以解决嘛？

正如前面所说，如果是格式检查，请明说。
https://github.com/stone-zeng/fduthesis/blob/d84e0dd7df657d47943e98c98ff843cc1336bc5f/source/fduthesis.dtx#L4422

**Reply by yaoyichloe** [reply]
是的，是格式检查提出的问题，格式检查还要求给伪代码加表头。
但是我加了表头后字体和间距都不太好看，可能是我加的命令不好，请问有什么解决方案吗

**Reply by yaoyichloe** [reply]
\captionsetup [ figure ] { font = small,         labelsep = quad }
\captionsetup [ table  ] { font = { small, sf }, labelsep = quad }
还想请问下，您标注的这两句话，是改动什么的呀？我应该加在什么位置呀

**Reply by sikouhjw** [reply]
> 是的，是格式检查提出的问题，格式检查还要求给伪代码加表头。 但是我加了表头后字体和间距都不太好看，可能是我加的命令不好，请问有什么解决方案吗

算法的 caption 应当使用算法宏包提供的接口，例如 `algorithm2e` 宏包：
![图片](https://github.com/stone-zeng/fduthesis/assets/49770925/4426802e-519a-4b61-8627-3239783e799f)

> \captionsetup [ figure ] { font = small, labelsep = quad } \captionsetup [ table ] { font = { small, sf }, labelsep = quad } 还想请问下，您标注的这两句话，是改动什么的呀？我应该加在什么位置呀

其中 `table/font/sf` 的意思是使用无衬线字体，你不需要无衬线，那么在 cls 中删除 `sf` 或者在导言区添加
```tex
\captionsetup [ table ] { font = { small, rm } }
```
